aspnetcore/host-and-deploy/azure-apps/troubleshoot.md

```diff
@@ -5,12 +5,203 @@ description: Learn how to diagnose problems with ASP.NET Core Azure App Service
+
+## Monitoring blades
+
+Monitoring blades provide an alternative troubleshooting experience to the methods described earlier in the topic. These blades can be used to diagnose all 500-series errors.
+
+Confirm that the ASP.NET Core Extensions are installed. If the extensions aren't installed, install them manually:
+
+1. In the **DEVELOPMENT TOOLS** blade section, select the **Extensions** blade.
+1. The **ASP.NET Core Extensions** should appear in the list.
+1. If the extensions aren't installed, select the **Add** button.
+1. Choose the **ASP.NET Core Extensions** from the list.
+1. Select **OK** to accept the legal terms.
+1. Select **OK** on the **Add extension** blade.
+1. An informational pop-up message indicates when the extensions are successfully installed.
+
+If stdout logging isn't already enabled, follow these steps:
+
+1. In the Azure portal, select the **Advanced Tools** blade in the **DEVELOPMENT TOOLS** area. Select the **Go&rarr;** button. The Kudu console opens in a new browser tab or window.
+1. Using the navigation bar at the top of the page, open **Debug console** and select **CMD**.
+1. Open the folders to the path **site** > **wwwroot** and scroll down to reveal the *web.config* file at the bottom of the list.
+1. Click the pencil icon next to the *web.config* file.
+1. Set **stdoutLogEnabled** to `true` and change the **stdoutLogFile** path to: `\\?\%home%\LogFiles\stdout`.
+1. Select **Save** to save the updated *web.config* file.
+
+Proceed to activate diagnostic logging:
+
+1. In the Azure portal, select the **Diagnostics logs** blade.
+1. Select the **On** switch for **Application Logging (Filesystem)** and **Detailed error messages**. Select the **Save** button at the top of the blade.
+1. To include failed request tracing, also known as Failed Request Event Buffering (FREB) logging, select the **On** switch for **Failed request tracing**. 
+1. Select the **Log stream** blade, which is listed immediately under the **Diagnostics logs** blade in the portal.
+1. Make a request to the app.
+1. Within the log stream data the cause of the error is indicated.
+
+**Important!** Be sure to disable stdout logging when troubleshooting is complete:
+
+1. In the Kudu **Diagnostic Console**, return to the path **site** > **wwwroot** to reveal the *web.config* file. Open the **web.config** file again by selecting the pencil icon.
+1. Set **stdoutLogEnabled** to `false`.
+1. Select **Save** to save the file.
+
+For example, the following log stream message indicates that the app's assembly name (*troubleshoo.dll*) is incorrect. The assembly name should be *troubleshoot.dll*:
```

Comment:
Can you replace this with exception? That error is just too weird.

Reply:
Let me know which error you want. I wasn't considering "weird." I was considering *obvious*. I wanted something so simple to understand that the reader won't dwell on it. This error is very simple ... `dotnet trobleshoo.dll` ... that just ain't gonna fly.

Reply:
Do the error I did in the video. 

throw("The seed method failed"); or something like that.

Reply:
If ur making me do it, ok. :smile: That's *not* a real startup error. I really think that if we timed a reader reading what's there now and reading that exception that it will slow the reader down thinking about that. However, that depends on what the 502.5 message is in the event log.

I'll run it here, since I've never purposely thrown in Main before like that. You've got me curious now to see exactly what the error message looks like.

Reply:
Then pick a better startup error and throw it. You can always throw("My sample startup exception.") if you think that's better.

Reply:
I think what's better is that we use any of the *real* startup errors that produce the 'I can't start the process' error message in the log. Of all of them, having the wrong assembly name is the rock-bottom simplest and easiest to understand ... just misspelling the assembly name will do it.

Reply:
I have another idea ... let's try the next version with the image of the error but not the specific error description.

The important thing at this point in the topic is that they check the App Event Log. If there's an error, they see if it's helpful or not. If it is ... whatever it is ... they just act on it. If it isn't helpful (e.g., there is no error or the error is cryptic), then they just move on to the Kudu console run.

I'm concerned about bogging the reader down in the example anyway. I'd rather have them just look :eyes: and then get on with it.